crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -108,6 +108,32 @@ impl<T: TransactionOrdering> TxPool<T> {
         self.all().txs_iter(sender).last().map(|(_, tx)| Arc::clone(&tx.transaction))
     }
 
+    /// Returns the next valid transaction for the given sender.
```

Comment:
going by the test, this returns the highest consecutive nonce in the pool, right? so the consumer would need to +1 on the value returned by this fn. this is slightly different behavior than e.g. the nonce RPC API, as this actually returns the exact value you need (i.e. no need to +1)